src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.Arrays;
```

Comment:
The use of `Arrays` utility class is not allowed as per the checklist. Consider implementing array copying manually or using another approach.